src/block/RedMushroom.php

```diff
@@ -43,7 +43,12 @@ public function onNearbyBlockChange() : void{
 
 	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
 		$down = $this->getSide(Facing::DOWN);
-		if(!$down->isTransparent()){
+		$position = $this->getPosition();
+		$lightLevel = $position->getWorld()->getFullLightAt($position->x, $position->y, $position->z);
+		$highestBlock = $position->getWorld()->getHighestBlockAt($position->x, $position->z);
```

Comment:
This is redundant. If the light level is < 13, the block cannot be under direct sky light. In addition, placement of blocks such as glass don't bypass this limit.